lib/cloudinary.rb

```diff
@@ -86,21 +86,27 @@ def self.config(new_config=nil)
 
   def self.config_from_url(url)
     @@config ||= OpenStruct.new
-    uri      = URI.parse(url)
-    set_config(
-      "cloud_name"          => uri.host,
-      "api_key"             => uri.user,
-      "api_secret"          => uri.password,
-      "private_cdn"         => !uri.path.blank?,
-      "secure_distribution" => uri.path[1..-1]
-    )
-    uri.query.to_s.split("&").each do
-    |param|
-      key, value = param.split("=")
-      if isNestedKey? key
-        putNestedKey key, value
-      else
-        set_config(key => URI.decode(value))
+    if url
```

Comment:
Best to use a guard clause here:

```suggestion
    return unless url && !url.empty?
```

Reply:
Updated